crates/blobstream/src/blobstream.rs

```diff
@@ -145,8 +162,16 @@ pub fn verify_data_commitment_storage(
     expected_with_prefix.push(0xa0); // Add the RLP prefix
     expected_with_prefix.extend_from_slice(expected_commitment.as_slice());
 
+    // verify the value of the storage slot, then of the storage proof against
     match verify_proof(root, nibbles, Some(expected_with_prefix), &storage_proof) {
-        Ok(_) => Ok(()),
+        Ok(_) => {
+            let nibbles = Nibbles::unpack(keccak256(expected_blobstream_address));
+            let expected = alloy_rlp::encode(expected_blobstream_address);
+            match verify_proof(state_root, nibbles, Some(expected), &account_proof) {
```

Comment:
Should be verifying account proof with nibbles against expected account state not the rlp encoded blobstream address.